packages/frontend/core/src/modules/infra-web/global-scope/index.tsx

```diff
@@ -0,0 +1,44 @@
+import type { Page } from '@toeverything/infra';
+import {
+  LiveData,
+  ServiceCollection,
+  type ServiceProvider,
+  ServiceProviderContext,
+  useLiveData,
+  useService,
+  useServiceOptional,
+} from '@toeverything/infra';
+import type React from 'react';
+
+import { CurrentPageService } from '../../page';
+import { CurrentWorkspaceService } from '../../workspace';
+
+export const GlobalScopeProvider: React.FC<
+  React.PropsWithChildren<{ provider: ServiceProvider }>
+> = ({ provider: rootProvider, children }) => {
+  const currentWorkspaceService = useService(CurrentWorkspaceService, {
+    provider: rootProvider,
+  });
+
+  const workspaceProvider = useLiveData(
+    currentWorkspaceService.currentWorkspace
+  )?.services;
+
+  const currentPageService = useServiceOptional(CurrentPageService, {
+    provider: workspaceProvider ?? ServiceCollection.EMPTY.provider(),
+  });
+
+  const pageProvider = useLiveData(
+    currentPageService?.currentPage ?? new LiveData<Page | null>(null)
+  )?.services;
+
+  console.log('global', pageProvider ?? workspaceProvider ?? rootProvider);
```

Comment:
Not sure if it is better to handle all of these in rxjs

Reply:
What do you mean, provide an example?

Reply:
Something like this, define a smart service provider:

```ts
class SmartServiceProvider {
  constructor(
    private readonly currentWorkspaceService: CurrentWorkspaceService,
    private readonly provider: ServiceProvider
  ) {}

  value = LiveData.from(
    this.currentWorkspaceService.currentWorkspace.pipe(
      switchMap(workspace => {
        const workspaceProvider = workspace?.services;
        const pageService = workspaceProvider?.get(CurrentPageService, {
          optional: true,
        });
        const currentPage = pageService?.currentPage;
        return (currentPage ?? new LiveData<Page | null>(null)).pipe(
          map(page => {
            return page?.services ?? workspaceProvider ?? this.provider;
          })
        );
      })
    ),
    this.provider
  );
}
```

and then `useService(SmartServiceProvider).value`

Reply:
Thank you for your suggestion, but after yesterday's discussion of the multi-tab mode.
I feel that the design of this global scope is not very practical. In the future, each part on the page will have its own scope. global scope may be removed in the future.